### DISCUSSION TITLE
How should access the auth service in the model

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1386] [Help]
I have the model like below and have a relation for get the price of the model for user with loyalty level of the user

I had a model like below. but I can not access the `auth` service in the model.

```javascript
class Post {

    price()
    {
        const level = auth.user.loyalty_level;

       return this.hasOne('App/Models/Price').query().where('loyalty_level', level);
    }
}
```

is there any way to access the `auth` in the model?

**Comment by McSneaky** [comment]
You need to pass it to model function

Since model is not directly in http context there is no direct way to just pull it

**Reply by farshadfahimi** [reply]
In this way in each controller I should assign the user to the model and it can make bug in big system with team.

it should be a way to use `auth` out of the box. not with the request or just in the controller

for example in laravel you can use `Auth` class in every class or model you want

**Reply by thetutlage** [reply]
What do you suggest? How should we achieve it?

**Reply by McSneaky** [reply]
Laravel can do it, because that's how PHP works. But PHP and NodeJS are quite the different.. In NodeJS when you make global var it's global across all requests. In PHP gloabls are per request, which allows to use such pattern. 
Super simplified: PHP runs 1 request on 1 thread, while NodeJS runs all requests on single thread

**Comment by farshadfahimi** [comment]
@thetutlage 

in this situation like me. I think just the user is necessary to be accessible in all of the application to be used in `model`, `traits` or maybe some other classes.

It should be a good idea that `auth` service has an optional `middleware` that assign `auth user` to the gloabal variable like `Auth` and it accessible in all of the application and don't need to do some other stuff, and user just add it or not.

or maybe in config we could have a variable can be true or false and user can set want to access user globally or not

**Reply by RomainLanz** [reply]
I believe you have a misconception on how Node.js works.

Since Node.js is asynchronous, you cannot have a global variable that is not shared between requests. It means if you set up a global variable `Auth`, it will be shared across all requests and therefore, will not work.

One way of doing it is to use the AsyncStorage and there's a [PR for that](https://github.com/adonisjs/http-server/pull/18).

**Comment by thetutlage** [comment]
As @RomainLanz and @McSneaky  has mentioned, that is not how Node.js works. I suggest learning about Node.js first before opting for a framework. The basic understand will help you a lot
